packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts

```diff
@@ -69,27 +69,30 @@ export class AuthenticatorComponent
 
     const { authService, initializeMachine } = this.authenticator;
 
-    this.unsubscribeHub = listenToAuthHub(authService, (data, service) => {
-      defaultAuthHubHandler(data, service);
-      /**
-       * Hub events aren't properly caught by Angular, because they are
-       * synchronous events. Angular tracks async network events and
-       * html events, but not synchronous events like hub.
-       *
-       * On any notable hub events, we run change detection manually.
-       */
-      this.changeDetector.detectChanges();
-
-      /**
-       * Hub events that we handle can lead to multiple state changes:
-       * e.g. `authenticated` -> `signOut` -> initialState.
-       *
-       * We want to ensure change detection runs all the way, until
-       * we reach back to the initial state. Setting the below flag
-       * to true to until we reach initial state.
-       */
-      this.isHandlingHubEvent = true;
-    });
+    this.unsubscribeHub = listenToAuthHub(
+      authService,
+      async (data, service) => {
+        await defaultAuthHubHandler(data, service);
```

Comment:
Note that `defaultAuthHubHandler` is async now, to have better handling for race conditions (see #2760). This is not breaking because `authHubHandler` is a fire and forget side effect from Hub events anyway.

Reply:
👍